eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/producer/SendMessageContext.java

```diff
@@ -123,13 +123,13 @@ public String toString() {
     }
 
     @Override
-    public boolean retry() throws Exception {
+    public void retry() throws Exception {
         if (eventMeshProducer == null) {
-            return false;
+            log.error("Exception happends during retry. EventMeshProduceer is null.");
```

Comment:
here need add return in if block

Reply:
I was merging 3 `RetryContext`s in module http, grpc and tcp to one.
![1683297249354](https://user-images.githubusercontent.com/35672972/236488064-32bcba14-2deb-403e-9de1-2ea9ed9bd22b.png)
3 `RetryContext`s have almost same structure except a little difference. This `retry()` is that difference. It returns boolean in module http, grpc, and void in module tcp. Then I checked all calls of `retry()` that returns boolean. I found the return value of boolean has no practical use in all calls. So I chose to modify it to `void` to keep with `retry()` in module tcp.

If the return value of boolean type is indeed useful, may I modify the void to boolean in module tcp?

Reply:
What I mean is that in your modified logic, the return is missing in the if code block, and return boolean is not required.
you need add `return` after the `log.error()`.

Reply:
Oh! Yes. :joy: